Python/libraries/recognizers-text/recognizers_text/Matcher/IMatcher.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
+from typing import TypeVar
+
+
+class IMatcher(ABC):
+
+    T = TypeVar('T')
```

Comment:
I am not sure if we can manage the <T> (generic type in C#) just as a [str], instead of this approach. @Aliandi it is that correct?